tensorflow/lite/kernels/internal/reference/reference_ops.h

```diff
@@ -274,21 +274,19 @@ inline void QuantizeLeakyRelu(const LeakyReluParams& params, T q_alpha,
   const int flat_size = MatchingFlatSize(input_shape, output_shape);
   static const int32 quantized_min = std::numeric_limits<T>::min();
   static const int32 quantized_max = std::numeric_limits<T>::max();
-  static const int32 alpha_value = q_alpha - params.alpha_offset;
+  static const int32 alpha_value = params.q_alpha - params.alpha_offset;
+  static const int32 identity_value = params.q_identity - params.alpha_offset;
   for (int i = 0; i < flat_size; ++i) {
     const int32 input_value = input_data[i] - params.input_offset;
-    if (input_value >= 0) {
-      output_data[i] = input_data[i];
-    } else {
-      const int32 unclamped_output =
-          params.output_offset + MultiplyByQuantizedMultiplierSmallerThanOneExp(
-                                     input_value * alpha_value,
-                                     params.output_multiplier,
-                                     params.output_shift);
-      const T clamped_output =
-          std::min(quantized_max, std::max(quantized_min, unclamped_output));
-      output_data[i] = static_cast<uint8>(clamped_output);
-    }
+    auto q_mutliplier = (input_value >= 0) ? identity_value : alpha_value;
+    const int32 unclamped_output =
+        params.output_offset + MultiplyByQuantizedMultiplierSmallerThanOneExp(
```

Comment:
again, why you need to change the implementation?

Reply:
This piece of code implements the quantized inference of Leaky ReLU.

The problem of the original implementation is that when `input_value >= 0`, the quantized input was directly transferred to output. This seems OK if the input scale and output scale are the same, since the UINT8/INT8 encoding will be the same if they represent the same float number. However, if the input scale and output scale are different, then the same UINT8/INT8 encoding will represent different float value between input and output. 

Therefore, it is important to transform the quantized input to the quantized output in a proper way. It is similar to how the negative part was implemented. The only difference is that the negative part has `alpha_value` multiplied. In the positive part, we need to multiply the input by quantized identity, which was passed into the function as `params.q_identity`.

The original test was implemented in a way such that input and output scale are different, thus it didn't catch the error.